Varia: Add Color Annotations

allancole opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request: 

This PR attempts to add color-annotation support to Varia.

#### Screenshot: 

![varya-color-annotations-1](https://user-images.githubusercontent.com/709581/80406258-c4b43000-8891-11ea-988c-a96bd6432bc2.gif)

#### ToDo: 

- [x] Add support for Primary, Secondary, Foreground, Background, and Accent colors where approriate.
- [x] Improve the color-annotations plugin to support the Gutenberg editor styles and color palettes.
- [x] Add featured color palettes to Varia.
- [x] Apply this solution to all Varia-based child themes.

allancole committed:
Varia: Adding Color Annotations.

allancole committed:
Varia: Remove `!important` from utility color rules in frontend styles.

allancole committed:
Varia: Various color annotation improvements.

allancole committed:
Varia: Improve color annotations to deal with form inputs and theme border colors.

allancole commented:
Still working through all the subtle details but after the last batch of commits here’s where things are: 

![varya-color-annotations-2](https://user-images.githubusercontent.com/709581/80423370-ec18f600-88ad-11ea-90be-04dcbe5b36c0.gif)

Works pretty well so far and it looks like we’ll be able to make and custom-colors configuration a11y friendly.

allancole committed:
Varia: Improving utility class integration, add text-selection color support.

allancole committed:
Varia: Improving utility class annotations. Revising selection colors.

allancole committed:
Varia: More code tidying to add better comments and break out utility classes.

kjellr reviewed (commented):
This looks pretty good! I tested it out and had only one really minor observation: 

I think the SVG icons here need a `fill: curentColor` rule applied. Might as well bundle that into this PR. 

![Screen Shot 2020-04-29 at 12 40 14 PM](https://user-images.githubusercontent.com/1202812/80622746-39fd3d80-8a17-11ea-81fc-dbcc46725039.png)

_EDIT: I pushed a quick fix for this in_ 1151c03

kjellr committed:
Ensure next/previous arrows in post navigation can change color too.

allancole committed:
Varia: Adding custom color values to editor palettes.
- Keeping the grey variant colors to prevent disruptions in existing sites.

allancole committed:
Varia: Adding a new, separate color-annotation for the editor.

kjellr commented on varia/functions.php:
```diff
@@ -145,19 +145,25 @@ function varia_setup() {
 		// $lightness       = varia_get_default_lightness();
 		// $lightness_hover = varia_get_default_lightness_hover();
 
+		$colors_array = get_theme_mod( 'colors_manager' );
```
Minor, but can you add a code comment explaining what this is?

allancole replied:
Added some comments in a28de42

kjellr replied:
Thanks!

allancole committed:
Varia: Remove unnecessary editor colors. Code tidying.

allancole committed:
Varia: Clean up conditional statment.

allancole committed:
Varia: Code tidying.

allancole committed:
Varia: Improving editor color annotations.

allancole committed:
Varia: Remove contrast rules in colors annotations for simpler color controls.

allancole committed:
Varia: Add opacity rule to dim some text colors.

allancole committed:
Varia: fix border width issue in separator block.

allancole committed:
Varia: cleaning up annotations against wpcom.

allancole committed:
Varia: Improvements to custom colors
- Add a11y color note to customizer
- Remove image filter function.
- Improve selection colors
- Code tidying

kjellr commented on varia/style.css:
```diff
@@ -2436,35 +2440,35 @@ table th,
 }
 
 .has-primary-color[class] {
-	color: blue !important;
```
Will the removal of these `!important` rules automatically flow down to the child themes when we recompile their CSS? It doesn't look like these rules are defined in the child theme CSS in the first place.

kjellr committed:
Add featured color palettes

kjellr committed:
Remove custom ::selection rules.

This causes bold selection colors... probably too bold.

allancole committed:
Varia: Remove custom selection colors, add new Varia color palettes for light, medium and dark palettes.

allancole committed:
Merge branch 'try/add-color-annotations' of https://github.com/Automattic/themes into try/add-color-annotations

allancole committed:
Varia: Code tidying.

allancole committed:
Varia: Add learn more link to color contrast message.

kjellr commented on varia/functions.php:
```diff
@@ -140,10 +140,16 @@ function varia_setup() {
+		 * - if the customizer color is empty, use the default
+		 */
+		$colors_array = get_theme_mod('colors_manager'); // color annotations array()
+		$background   = $colors_array['colors']['bg'];   // $config-global--color-background-default;
```
One thing I noticed is that this throws errors when I run the theme locally. For example: 

```
Notice: Trying to access array offset on value of type bool in ../wp-content/themes/dotcom/alves/functions.php on line 62

Notice: Trying to access array offset on value of type null in ../wp-content/themes/dotcom/alves/functions.php on line 62
```

Can we wrap these in a conditional to prevent that?

allancole replied:
Figured out a solution for this and revised the the conditionals in all the child themes so they don’t throw an error on non-dotcom setups, and so they’re a little easier to digest. Also switched to `! empty()` from `! isset()` which is a better [kind of check](https://phppot.com/php/isset-vs-empty-vs-is_null/).

allancole committed:
Varia: Revise conditionals to fix array offset issue, cody tidying.

allancole committed:
Varia: Code tidying.

allancole committed:
Varia: Remove `!important` flags from utility classes and recompile.

allancole committed:
Varia: Fix nested button text-color.

allancole committed:
Varia: Cleaning up table color-annotations

allancole committed:
Varia: Clean up table color-annotations

allancole committed:
Varia: Improve color annotations and Gutenberg 5.4.2 support.
- Add support for table block and associated options
- Improve color annotations for tabel block
- Cleanup button, cover, and utility styles
- Recompile style sheet

allancole committed:
Merge branch 'try/add-color-annotations' of https://github.com/Automattic/themes into try/add-color-annotations

allancole commented:
Just adding a not that this PR fixes #2103

allancole committed:
Varia: Clean up cover and utility styles for dim.

allancole committed:
Varia: Clean up nested text-color specificity.

allancole commented:
Merging this PR and syncing the changes to wpcom.
